be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/GetMessagesByIdResponseHandlerSuite.scala

```diff
@@ -1,99 +1,85 @@
     }
-    def receive: Receive = counter(0)
   }
 
-  private val testProbe = TestProbe()
-  final val CHANNEL1: Channel = Channel("/root/wex/lao1Id")
-  final val CHANNEL2: Channel = Channel("/root/wex/lao2Id")
-  final val MESSAGE1_ID: Hash = Hash(Base64Data.encode("message1Id"))
-  final val MESSAGE2_ID: Hash = Hash(Base64Data.encode("message2Id"))
-  final val MESSAGE1: Message = Message(null, null, null, MESSAGE1_ID, null, null)
-  final val MESSAGE2: Message = Message(null, null, null, MESSAGE2_ID, null, null)
-  final val missingMessages = Map((CHANNEL1, Set(MESSAGE1)), (CHANNEL2, Set(MESSAGE2)))
-  final val receivedResponse: JsonRpcResponse = JsonRpcResponse(RpcValidator.JSON_RPC_VERSION, new ResultObject(missingMessages), None)
-
-  test("by receiving a get_messages_by_id response, it sends a write message to the database") {
-    val boxUnderTest: Flow[GraphMessage, GraphMessage, NotUsed] = GetMessagesByIdResponseHandler.graph(system.actorOf(Props(new TestDb(testProbe.ref))))
-    val input: List[GraphMessage] = List(Right(receivedResponse))
-    val source = Source(input)
-    val s = source.via(boxUnderTest).runWith(Sink.seq[GraphMessage])
-    Await.ready(s, duration)
-    testProbe.expectMsgAllOf((CHANNEL1, MESSAGE1), (CHANNEL2, MESSAGE2))
-    testProbe.expectNoMessage()
-  }
+  test("After processing a correct get_messages_by_id response the db should contain all the channels and messages from it") {
 
-  test("by failing to write a  get_messages_by_id response in the database, it retries exactly three times before giving up") {
-    val boxUnderTest: Flow[GraphMessage, GraphMessage, NotUsed] = GetMessagesByIdResponseHandler.graph(system.actorOf(Props(new FailingTestDb(testProbe.ref))))
-    val input: List[GraphMessage] = List(Right(receivedResponse))
-    val source = Source(input)
-    val s = source.via(boxUnderTest).runWith(Sink.seq[GraphMessage])
-    Await.ready(s, duration)
-    testProbe.expectMsg((CHANNEL1, MESSAGE1))
-    testProbe.expectMsg((CHANNEL1, MESSAGE1))
-    testProbe.expectMsg((CHANNEL1, MESSAGE1))
-    testProbe.expectMsg((CHANNEL2, MESSAGE2))
-    testProbe.expectMsg((CHANNEL2, MESSAGE2))
-    testProbe.expectMsg((CHANNEL2, MESSAGE2))
-    testProbe.expectNoMessage()
-  }
+    val ask = dbActorRef ? DbActor.GetAllChannels()
+    val channelsInDb = Await.result(ask, duration) match {
+      case DbActor.DbActorGetAllChannelsAck(channels) => channels
+      case err @ _                                    => Matchers.fail("Error: " + err.toString)
+    }
+
+    val channelsInResponse = getMessagesByIdResponse.result.get.resultMap.get.keySet
+    channelsInResponse.diff(channelsInDb) should equal(Set.empty)
+
+    val messagesInResponse = getMessagesByIdResponse.result.get.resultMap.get.values.foldLeft(Set.empty: Set[Message])((acc, set) => acc ++ set)
+    val allMessages: Set[Message] = channelsInDb.foldLeft(Set.empty: Set[Message])((acc, channel) => acc ++ getMessages(channel))
```

Comment:
nitpick: for consistency with the variable `channelsInDb`, this variable could be renamed `messagesInDb` and be put in the line above